Schema.php

```diff
@@ -449,7 +449,35 @@ protected function getColumnPhpType($column)
      */
     protected function findColumns($index)
     {
-        $sql = 'DESCRIBE ' . $this->quoteSimpleIndexName($index->name);
+        $columns = $this->getIndexColumns($index->name);
+        if (!empty($columns[0]['Agent'])) {
+            $columns = $this->getIndexColumns($columns[0]['Agent']);
```

Comment:
THe original code was recursive; agent of the destributed index can contain its own 'Agent', which should be traversed. This patch breaks the recursion.